### ISSUE TITLE
Wall count doubles with modifiers over 3 walls

### ISSUE THREAD

**Issue by MatthaeusLeiner** [bambulab/BambuStudio#4664]
### Bambu Studio Version

1.9.3.50

### Where is the application from?

Bambu Lab Official website

### OS version

Windows 10

### Printer

X1E

### How to reproduce

Insert a Zylindrical Modifiers and up the wall count above 3. 

![image](https://github.com/user-attachments/assets/d4f84531-0ece-4102-8ab0-d74b54566dd5)

### Actual results

With 3 walls:

![image](https://github.com/user-attachments/assets/42cb2852-2580-4feb-bde5-4495c9b74a2c)

With 4 walls:

![image](https://github.com/user-attachments/assets/42d434d7-927b-4f66-9481-4411900b297f)

It also creates a gap on the outside wall with 4 walls:

![image](https://github.com/user-attachments/assets/53b962b2-4fb3-4a06-bf5f-9d5643655ddf)

### Expected results

It should have only the inserted wall amount.

### Project file & Debug log uploads

[Wallerror.zip](https://github.com/user-attachments/files/16707605/Wallerror.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

**Comment by QingZhangBambu** [comment]
yes, it does a problem. but a bit hard to fix. I will add it to our fix list.

**Comment by vanowm** [comment]
@zhimin-zeng-bambulab How is this a feature request? It's clearly a bug
